tests/std/tests/GH_002045_put_time_changes_errno/test.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cerrno>
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+using namespace std;
+using namespace chrono;
+
+int main() {
+    time_t t(system_clock::to_time_t(system_clock::now()));
+    {
+        ostringstream os;
+
+        assert(errno == 0);
+
+        std::tm tm;

```

Comment:
This test is `using namespace std;`, so giving the variable the same name as its type, and then using `std::` qualification, seems unnecessarily confusing. I'd recommend avoiding the shadowing with `tm_val`.